server/src/session.ts

```diff
@@ -377,13 +379,61 @@ export class Session {
     }
   }
 
+  /**
+   * Creates an external project with the same config path as `project` so that TypeScript keeps the
+   * project open when navigating away from `html` files.
+   */
+  private createExternalProject(project: ts.server.Project) {
+    if (isConfiguredProject(project) &&
+        !this.configuredProjToExternalProj.has(project.projectName)) {
+      const extProjectName = `${project.projectName}-external`;
+      project.projectService.openExternalProject({
+        projectFileName: extProjectName,
+        rootFiles: [{fileName: project.getConfigFilePath()}],
+        options: {}
+      });
+      this.configuredProjToExternalProj.set(project.projectName, extProjectName);
+    }
+  }
+
   private onDidCloseTextDocument(params: lsp.DidCloseTextDocumentParams) {
     const {textDocument} = params;
     const filePath = uriToFilePath(textDocument.uri);
     if (!filePath) {
       return;
     }
     this.projectService.closeClientFile(filePath);
+    this.maybeCloseExternalProject(filePath);
+  }
+
+  /**
+   * We open external projects for files so that we can prevent TypeScript from closing a project
+   * when it shouldn't be. This function checks if there are no longer any open files in the project
```

Comment:
... prevent TypeScript from closing a project when there is open external HTML template that still references the project.